subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -336,12 +336,9 @@ Using a task directly as an argument like this relies on it having <<more_about_
 [[sec:single_file_paths]]
 === Single file paths
 
-One of the great quandaries when developing a build is how to specify file locations when the build may be executed from an arbitrary directory — not necessarily in the project — and may be run on any number of different systems with incompatible directory layouts. The standard Java mechanism for specifying a file path runs into trouble in these situations:
+Never use `new File(relative path)`, as this creates a path relative to the current working directory, which could be anywhere.
 
- * `new File(relative path)` generates a path relative to the current working directory, which could be anywhere
- * `new File(absolute path)` will fail if the file system doesn't have the requisite path.
-
-Gradle solves this problem by providing the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)[Project.file(java.lang.Object)] method, which generates a path relative to the _project_ directory (unless the given path is absolute, in which case it is used as is). Here are some examples of using the `file()` method with different types of argument:
+Always use the  the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)[Project.file(java.lang.Object)] method, which generates a path relative to the _project_ directory (unless the given path is absolute, in which case it is used as is). Here are some examples of using the `file()` method with different types of argument:
```

Comment:
I would put this paragraph first and start it with:

> Gradle provides the `Project.file()` method for specifying the location of a single file or directory...

I would then consider putting the warning about `new File()` in a `[CAUTION]` block so that readers don't miss it.